js/components/thumbnail.js

```diff
@@ -7,12 +7,107 @@ var React = require('react'),
     Utils = require('./utils');
 
 var Thumbnail = React.createClass({
+  getInitialState: function() {
+    this.positionY = 0;
+    this.positionX = -320;
+    this.imageWidth = 0;
+    this.imageHeight = 0;
+    this.thumbnailWidth = 80;
+    this.thumbnailHeight = 40;
+    return {};
+  },
+  componentDidMount: function() {
+    this.setThumbnailSizes();
+    this.setImageSizes();
+    var yaw = this.props.vrViewingDirection.yaw;
+    var pitch = this.props.vrViewingDirection.pitch;
+    this.setCurrentViewVr(yaw, pitch);
+  },
+  componentWillReceiveProps: function(nextProps) {
+    if (this.props.vrViewingDirection !== nextProps.vrViewingDirection) {
+      var yaw = nextProps.vrViewingDirection.yaw;
+      var pitch = nextProps.vrViewingDirection.pitch;
+      this.setCurrentViewVr(yaw, pitch);
+    }
+  },
   shouldComponentUpdate: function(nextProps) {
-    return (nextProps.hoverPosition != this.props.hoverPosition);
+    var hoverPosition = nextProps.hoverPosition != this.props.hoverPosition;
```

Comment:
Can you rename these variable to updateHoverPositon, updateFullscreen, updateVrViewDIrection. It's a little bit clearer.

Reply:
\+